src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,336 @@
+             r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         false);
+             SOL_INT_CHECK_GOTO(r, < 0, error);
+        }
+
+        if (current_state == SOL_NETCTL_SERVICE_STATE_ONLINE) {
+              r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         true);
+             SOL_INT_CHECK_GOTO(r, < 0, error);
+        }
+
+        mdata->state = current_state;
+
+        return;
+    }
+
+error:
+    mdata->state = current_state;
+
+    SOL_WRN("service %s failed to send online state", name);
+}
+
+void
+error_cb(void *data, const struct sol_netctl_service *service,
+    unsigned int error)
+{
+    struct sol_flow_node *node = data;
+    const char *name;
+
+    name = sol_netctl_service_get_name(service);
+
+    if (name)
+        sol_flow_send_error_packet(node, ENOENT, "Service %s error is %d", name, error);
+}
+
+static struct sol_netctl_service *find_service_by_name(char *service_name)
+{
+    const struct sol_ptr_vector *service_list;
+    struct sol_netctl_service *service;
+    const char *name;
+    uint16_t i;
+
+    service_list = sol_netctl_get_services();
+
+    SOL_PTR_VECTOR_FOREACH_IDX(service_list, service, i) {
+        name = sol_netctl_service_get_name(service); 
+        if (name && strcmp(name, service_name) == 0)
+            return service;
+    }
+
+    return NULL;
+}
+
+static int
+connection_handle(struct sol_flow_node *node, void *data, bool connect)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+
+    if (connect) {
+       r = sol_netctl_service_connect(service);
+       if (r < 0)
+            sol_flow_send_error_packet(node, -r, "Could not connect to service: %s: %s",
+                     mdata->service_name, sol_util_strerrora(-r));
+    } else {
+       r = sol_netctl_service_disconnect(service);
+       if (r < 0)
+            sol_flow_send_error_packet(node, -r, "Could not disconnect to service: %s: %s",
+                     mdata->service_name, sol_util_strerrora(-r));
+    }
+
+    return 0;
+}
+
+static int
+connect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, true);
+}
+
+static int
+disconnect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, false);
+}
+
+static int
+get_service_state(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
```

Comment:
get_service_state is not totally similar with service_status_cb. 
In service_status_cb, it wil emit online or offline when the status changed. So I use the current_state compared with previous state.
In get_service_state , it will emit online or state by Trigger node, so it always emit the current_state by given a service.
So this part cannot merged into one new function. In service_status_cb there are some logic to compare current state and previous state.  In get_service_state ,just emit current status packet.